### PR TITLE
Resolve #9 - Add custom screenshot paths

### LINKED ISSUE #9: Generate better name for screenshot file.
Currently `perceptiactl` saves screenshot to `screenshot.png`. The file name should also contain current time to prevent overriding files. It should be also possible to pass name through program arguments.

TIP: `perceptiactl` is simple diagnostic tool. Command `parceptiactl screenshot` will take screenshot.

### PR DESCRIPTION
**Pull request by zottce**

### CODE REVIEW

File: perceptia/perceptiactl/perceptiactl.rs [hunk 1]
```diff
@@ -49,8 +55,13 @@ fn main() {
...
             about::process();
         }
-        ("screenshot", Some(_)) => {
-            screenshot::process();
+        ("screenshot", Some(subcommand)) => {
+            if let Some(path) = subcommand.value_of("path") {
+                screenshot::process(String::from(path));
+            } else {
+                let now = chrono::Local::now();
+                screenshot::process(now.format("%Y-%m-%d_%H%M%S").to_string());
```
Comment (darkelement) [inline perceptia/perceptiactl/perceptiactl.rs]: This file name will not work. It has to have extension. Without it `image` crate fails to determine the format. What do you think about adding some prefix? "screenshot-%Y-%m-%d_%H%M%S.png" would be good.

File: perceptia/perceptiactl/screenshot.rs [hunk 2]
```diff
@@ -19,26 +19,30 @@ const GLOBAL_SCREENSHOOTER: &'static str = "weston_screenshooter";
...
+struct ScreenshooterConstructor {
+    path: String,
+}
 
 impl ScreenshooterConstructor {
-    fn new() -> Self {
-        ScreenshooterConstructor {}
+    fn new(path: String) -> Self {
+        ScreenshooterConstructor {
+            path,
```
Comment (darkelement) [inline perceptia/perceptiactl/screenshot.rs]: Some distributions are not up to date and still use rustc 1.16 where field shorthands give an error.

**Review by proton-decay — CHANGES_REQUESTED** [review]
Thank you! Two small corrections and this commit will be perfect.

**Comment by ZOTTCE** [comment]
Oh, sorry, forgot about an extension. It seems good now.

**Review by proton-decay — APPROVED** [review]

**Comment by proton-decay** [comment]
Now it works perfectly. Do you maybe plan to stay longer with perceptia? If so, do you see other interesting issues you would like to solve or should I find another ones? Do you have comments about perceptia?

**Comment by zottce** [comment]
I need to understand how perceptia works before I can start to do something.

**Comment by proton-decay** [comment]
Small issues are great for learning.

Here's general overview of architecture: https://github.com/perceptia/perceptia/wiki/Architecture

If something is not obvious by reading the code (and that's probably the case in many places) - this is a bug. Open an issue and I will add more comments or documentation.

Soon there will be support for offscreen test mode with VNC so it will be easier to develop perceptia without second computer.

**Pull request merged**